teps/0057-windows-support.md

```diff
@@ -0,0 +1,154 @@
+In order to have Windows nodes run TaskRuns there are some elements of Tekton that will need reworking to be compatible with that OS. Some of these have been identified in the [issue](https://github.com/tektoncd/pipeline/issues/1826) but it is almost certain that other elements will need work in order to work. 
+
+### Risks and Mitigations
+- The burden of supporting the Windows platform may be quite high
+- Windows containers may introduce some security vulnerabilities
+
+### User Experience (optional)
+- User specifies via nodeSelector if they want to run a task in a Windows container
+    - It just works
+- The default (no nodeSelector specified) behaviour selects a Linux container
+    - No noticeable change for users who use Linux only TaskRuns.
+
+### Performance (optional)
+In the case where there are no Windows containers the goal is obviously no performance impact at all. When there are Windows containers then we would aim for equivalent performance as on Linux nodes, where possible.
+
+## Design Details
+TBD
+
+## Test Plan
+TBD
+
+## Design Evaluation
+### Reusability
+Windows support is definitely a feature that cannot be achieved using any existing features or by adding something to the catalog. However, the way in which we add Windows support cannot be limited to our unique use case. This nodeSelector functionality may provide a pathway for other platforms (e.g. if OSX containers ever become feasible).
+### Simplicity
+Windows support needs to work with everything else already in Tekton. We’d also need to examine our design and implementation plans carefully to see if there are edge cases which would add unnecessary complexity that we can avoid.
+### Flexibility
+As described above, Windows support would be explicitly opt-in, meaning that non-Windows users never need to know about this feature. Adding Windows support would actually improve the flexibility of Tekton significantly by supporting workflows that are dependent on Windows containers.
+### Conformance
+Other than having to specify that a given TaskRun needs to run on a Windows node there really shouldn’t be any significant user-facing changes. With that in mind, we intend that the user experience when creating TaskRuns for Windows nodes would be identical to the experience when creating Linux based TaskRuns, other than using nodeSelector to specify the need for a Windows node.
+
+## Drawbacks
+- Support burden for Tekton community when Windows is supported this opens a huge set of new usages.
+- Future features may be somewhat hamstrung if there is a goal to provide parity between Windows/Linux support in Tekton.
+
+## Alternatives
+- Running a Windows VM inside a Linux container. This is a cumbersome alternative and has performance issues that make it untenable in production environments.
+
+## Infrastructure Needed (optional)
+We can work in a Fork of Tekton or the Tekton experiments. Whichever the Tekton contributors consider the better alternative.
```

Comment:
we're also going to need (in the long run) a way to test on Windows - which means we'll need to add that to our existing test infra (which is totally reasonable! but worth calling out i think :D)